rest_framework/schemas.py

```diff
@@ -467,31 +466,36 @@ def get_description(self, path, method, view):
         This will be based on the method docstring if one exists,
         or else the class docstring.
         """
-        method_name = getattr(view, 'action', method.lower())
-        method_docstring = getattr(view, method_name, None).__doc__
-        if method_docstring:
+        method = method.lower()
+        method_name = getattr(view, 'action', method)
+        view_method = getattr(view, method_name, None)
+        description = None
+        if view_method:
             # An explicit docstring on the method or action.
-            return formatting.dedent(smart_text(method_docstring))
-
-        description = view.get_view_description()
+            description = api_settings.VIEW_DESCRIPTION_FUNCTION(view_method)
```

Comment:
@xordoquy Was passing by and got interested by this PR. I took a look at the code and decided to chime in a bit.

Returning is certainly not an option here, because the whole point of this patch is to recognize sections of multi-method routes. So, it's what it does here, is doing the same as `view.get_view_description()` does for classes, but for route methods.

The weird part is, that in `description = view.get_description()` branch, a formatter (Markdown or whatever's configured) function would be [applied first](https://github.com/encode/django-rest-framework/blob/d300c3c/rest_framework/views.py#L235), and then the results will be parsed for sections (`splitlines`, then split to sections by `header_regex` etc etc). But this is how the code currently works, even before this patch.